src/test/java/com/devonfw/sample/archunit/DevonPackage/PackageStructure.java

```diff
@@ -0,0 +1,453 @@
+     *         to architecture pattern,
+     *         {@code false} otherwise.
+     * 
+     */
+    public boolean isValid() {
+
+        return this.valid;
+    }
+
+    /**
+     * @return the name of the root-package before the pattern matched. Will be the
+     *         empty {@link String} if not
+     *         {@link #isValid() valid}.
+     */
+    public String getRoot() {
+
+        return this.root;
+    }
+
+    /**
+     * @param otherPkg the other {@link PackageStructure} to compare.
+     * @return {@code true} if both this and the given
+     *         {@link PackageStructure}s have
+     *         the same {@link #getRoot() root},
+     *         {@code false} otherwise.
+     */
+    public boolean hasSameRoot(PackageStructure otherPkg) {
+
+        return getRoot().equals(otherPkg.getRoot());
+    }
+
+    /**
+     * @param otherPkg the other {@link PackageStructure} to compare.
+     * @return {@code true} if both this and the given
+     *         {@link PackageStructure}s have
+     *         the same {@link #getComponent()
+     *         component}, {@code false} otherwise.
+     */
+    public boolean hasSameComponent(PackageStructure otherPkg) {
+
+        return getComponent().equals(otherPkg.getComponent());
+    }
+
+    /**
+     * @param otherPkg the other {@link PackageStructure} to compare.
+     * @return {@code true} if both this and the given
+     *         {@link PackageStructure}s have
+     *         the same {@link #getComponent()
+     *         component}, {@link #getLayer() layer}, and {@link #getRoot() root},
+     *         {@code false} otherwise.
+     */
+    public boolean hasSameComponentPart(PackageStructure otherPkg) {
+
+        return hasSameComponent(otherPkg) && hasSameLayer(otherPkg) && hasSameRoot(otherPkg);
+    }
+
+    /**
+     * @return the name of the component. Will be the empty {@link String} if not
+     *         {@link #isValid() valid}.
+     */
+    public String getComponent() {
+
+        return this.component;
+    }
+
+    /**
+     * @return the name of the layer. Will be the empty {@link String} if not
+     *         {@link #isValid() valid}.
+     */
+    public String getLayer() {
+
+        return this.layer;
+    }
+
+    /**
+     * @param otherPkg the other {@link PackageStructure} to compare.
+     * @return {@code true} if both this and the given
+     *         {@link PackageStructure}s have
+     *         the same {@link #getLayer() layer},
+     *         {@code false} otherwise.
+     */
+    public boolean hasSameLayer(PackageStructure otherPkg) {
+
+        return getLayer().equals(otherPkg.getLayer());
+    }
+
+    /**
+     * @return {@code true} if the {@link #getLayer() layer} is a valid architecture
+     *         layer, {@code false} otherwise.
+     */
+    public boolean isValidLayer() {
+
+        return LAYERS.contains((this.layer));
+    }
+
+    /**
+     * @return {@code true} if the {@link #getLayer() layer} is {@link #LAYER_BATCH
```

Comment:
you can not link from public JavaDoc (of this method) to private JavaDoc (of field `LAYER_BATCH`).
Due to the blocker situation and urgency, I will quickly take over by doing a "constructive review" directly changing your code to avoid review loops and according delays. I hope that is fine. Will also add a JUnit as I found problems for some edge-cases.

Reply:
fixed it.